### PR TITLE
Rewrite overlay action

### PR DESCRIPTION
**Pull request by DPR00**
### Proposed changes

Rewrite overlay function following the guidelines in #441.

#### Type of change

- [ ] 🐛 Bugfix (change which fixes an issue)
- [x] 🚀 Feature (change which adds functionality)
- [ ] 📚 Documentation (change which fixes or extends documentation)

### Checklist

- [x] Lint and unit tests (if any) pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] All commits have been signed for [DCO](https://developercertificate.org/)

### CODE REVIEW

File: beluga/include/beluga/actions/overlay.hpp [hunk 1]
```diff
@@ -30,96 +30,90 @@ namespace beluga::actions {
...
-  using overlay_base_fn::operator();
 
-  /// Overload that defines a default execution policy.
   template <
-      class Range,
+      class ExecutionPolicy,
       class MaskRange,
       class Mask,
-      std::enable_if_t<ranges::range<Range>, int> = 0,
+      std::enable_if_t<std::is_execution_policy_v<ExecutionPolicy>, int> = 0,
```
Comment (nahueespinosa) [inline beluga/include/beluga/actions/overlay.hpp:101]:
```suggestion
      std::enable_if_t<std::is_execution_policy_v<std::decay_t<ExecutionPolicy>>, int> = 0,
```

File: beluga/include/beluga/actions/overlay.hpp [hunk 2]
```diff
@@ -30,96 +30,90 @@ namespace beluga::actions {
...
+///\cond detail
+
+template <class MaskRange, class Mask, class ExecutionPolicy = std::execution::sequenced_policy>
+struct overlay_closure {
+ public:
+  static_assert(std::is_execution_policy_v<ExecutionPolicy>);
+  static_assert(ranges::range<MaskRange>);
+
+  constexpr overlay_closure(MaskRange mask, Mask mask_value)
+      : policy_{std::execution::seq}, mask_(std::move(mask)), mask_value_(std::move(mask_value)) {}
```
Comment (nahueespinosa) [inline beluga/include/beluga/actions/overlay.hpp:42]: meta: The relationship between type name and variable name is a bit strange:
`MaskRange -> mask`
`Mask -> mask_value`

Consider these alternatives:
1. `MaskRange -> mask_range`
    `Mask -> mask`
2. `Mask -> mask`
    `MaskValue -> mask_value`
3. `MaskRange -> mask_range`
    `MaskValue -> mask_value`
    
And please propagate the change.

File: beluga/include/beluga/actions/overlay.hpp [hunk 3]
```diff
@@ -30,96 +30,90 @@ namespace beluga::actions {
...
+
+  constexpr explicit overlay_closure(ExecutionPolicy policy, MaskRange mask, Mask mask_value)
+      : policy_(std::move(policy)), mask_(std::move(mask)), mask_value_(std::move(mask_value)) {}
+
+  template <class Range>
+  constexpr auto operator()(Range& range) const -> Range& {
+    static_assert(ranges::range<Range>);
     auto map = range | ranges::views::common;
```
Comment (nahueespinosa) [inline beluga/include/beluga/actions/overlay.hpp:50]: nit:
```suggestion
    auto common_range = range | ranges::views::common;
```

File: beluga/include/beluga/actions/overlay.hpp [hunk 4]
```diff
@@ -30,96 +30,90 @@ namespace beluga::actions {
 
 namespace detail {
```
Comment (nahueespinosa) [inline beluga/include/beluga/actions/overlay.hpp:31]: nit: GitHub doesn't let me comment on the exact line, but I think this header is not needed here:
```cpp
#include <range/v3/view/transform.hpp>
```

File: beluga/include/beluga/actions/overlay.hpp [hunk 5]
```diff
@@ -30,96 +30,90 @@ namespace beluga::actions {
...
+struct overlay_closure {
+ public:
+  static_assert(std::is_execution_policy_v<ExecutionPolicy>);
+  static_assert(ranges::range<MaskRange>);
+
+  constexpr overlay_closure(MaskRange mask, Mask mask_value)
+      : policy_{std::execution::seq}, mask_(std::move(mask)), mask_value_(std::move(mask_value)) {}
+
+  constexpr explicit overlay_closure(ExecutionPolicy policy, MaskRange mask, Mask mask_value)
+      : policy_(std::move(policy)), mask_(std::move(mask)), mask_value_(std::move(mask_value)) {}
```
Comment (nahueespinosa) [inline beluga/include/beluga/actions/overlay.hpp:45]: nit: Use uniform initialization when possible.
```suggestion
      : policy_{std::move(policy)}, mask_{std::move(mask)}, mask_value_{std::move(mask_value)} {}
```

**Review by nahueespinosa — COMMENTED** [review]
@DPR00 Thanks for the patch! Looks good.

**Comment by DPR00** [comment]
I've just addressed the comments @nahueespinosa. Thanks for the help! 
@hidmic would be great if you could take a look too.

**Review by nahueespinosa — APPROVED** [review]
LGTM

**Review by hidmic — APPROVED** [review]
LGTM2

**Pull request merged**